File: type-script-nest-graph-ql-실습/src/app/equipment/model/dto/response/equipment-adv.response.dto.ts
import { EquipmentEntity } from "../../entities/equipment.entity";

export class EquipmentAdvResponseDto {
  private equipmentId: string;
  private usedBy: string;
  private count: number;
  private newOrUsed: string;
  private useRate: number;
  private isNew: boolean;

  constructor(equipment: EquipmentEntity) {
    this.equipmentId = equipment.equipmentId;
    this.usedBy = equipment.usedBy;
    this.count = equipment.count;
    this.newOrUsed = equipment.newOrUsed;

    if (equipment.usedBy === "developer") {
      this.useRate = Number(Math.random().toFixed(2));
    }

    this.isNew = equipment.newOrUsed === "new";
  }
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/equipment/model/dto/response/EquipmentAdvResponseDto.kt
package com.junyharangstudy.kotlingraphqltest.api.equipment.model.dto.response

class EquipmentAdvResponseDto (
    var equipmentId: String?,
    var usedBy: String?,
    var count: Int?,
    var newOrUsed: String?,
    var useRate: Float? = 0F,
    var isNew: Boolean? = false)